src/extensions/default/NavigationAndHistory/main.js

```diff
@@ -0,0 +1,464 @@
+                $("#mrof-container > #mrof-list > li > a.mroitem:visited").last().trigger("focus");
+            }
+            //_resetHideTimer();
+        } else {
+            _createMROFDisplayList();
+            $mrofContainer.addClass("confirmation-mode");
+            //_startHideTimer();
+            $(window).on("keyup", _hideMROFListOnAltlUp);
+        }
+    }
+
+    /**
+     * Opens the previous item in MROF list if pop over is visible else displays the pop over 
+     * @private
+     */
+    function _movePrev() {
+        var $context, $prev;
+        if ($mrofContainer) {
+            $context = $currentContext || $("#mrof-container > #mrof-list > li.highlight");
+            if ($context.length > 0) {
+                $prev = $context.prev();
+                if ($prev.length === 0) {
+                    $prev = $("#mrof-container > #mrof-list > li").last();
+                }
+                if ($prev.length > 0) {
+                    $currentContext = $prev;
+                    //_resetOpenFileTimer();
+                    $prev.find("a.mroitem").trigger("focus");
+                }
+            } else {
+                //WTF! (Worse than failure). We should not get here.
+                $("#mrof-container > #mrof-list > li > a.mroitem:visited").last().trigger("focus");
+            }
+            //_resetHideTimer();
+        } else {
+            _createMROFDisplayList();
+            $mrofContainer.addClass("confirmation-mode");
+            //_startHideTimer();
+            $(window).on("keyup", _hideMROFListOnAltlUp);
+        }
+    }
+    
+    /**
+     * Adds an entry to MROF list
+     * @private
+     * @param {Editor} editor - editor to extract file information
+     */
+    function _addToMROFList(editor) {
+        
+        var filePath = editor.document.file.fullPath;
+        
+        // Check existing list for this doc path and pane entry
+        var index = _.findIndex(_mrofList, function (record) {
+            return (record.file === filePath);
+        });
+
+        var entry = _makeMROFListEntry(filePath, editor._paneId, editor.getCursorPos(true, "first"));
+
+        if (index !== -1) {
+            _mrofList.splice(index, 1);
+        }
+
+        // add it to the front of the list
+        _mrofList.unshift(entry);
+    }
+    
+    EditorManager.on("activeEditorChange", function (event, current, previous) {
+        if (previous) {
+            _addToMROFList(previous);
+        }
+        
+        if (current) {
+            _addToMROFList(current);
+            _activePaneId = MainViewManager.getActivePaneId();
+        }
+    });
+    
+    ProjectManager.on("beforeProjectClose beforeAppClose", function () {
+        PreferencesManager.setViewState("openFiles", _mrofList, _getPrefsContext(), true);
+        _mrofList = [];
+    });
+    
+    ProjectManager.on("projectOpen", function () {
+        _mrofList = PreferencesManager.getViewState("openFiles", _getPrefsContext()) || [];
```

Comment:
I just realized I think you might want to preserve all files which are opened in the editor, but not in the working set. I think, it is OK to have a preference for this.